Terasology/MetalRenegades
# Water/Thirst system improvements and modifications

This pull request includes some major changes to the way that the thirst need is restored by players. These should make it so the thirst need is a greater challenge for players to fulfill.

<h3>All Changes</h3>

- [x] Wells can now be drunk from directly. This replenishes all of a player's thirst.
- [x] Instead of being given a water cup when interacting with a well, empty water cups can be bought from the market, and filled with a well.
- [x] Wells have a capacity of drinks, and the player will need to wait for a bit to drink again from this well when this limit is reached.
- [x] Tooltips have been included to indicate the status of wells.

![image](https://user-images.githubusercontent.com/44126397/86502347-c4964c00-bd7c-11ea-9b5c-186e8aa27086.png)
_A full well, and an empty well._

<h3>Testing</h3>

- Travel out in the world until a city spawns. Find a well (3 by 3 structure with water in the middle)
- Look at the water and press 'e' with nothing in your hand. Your thirst (the blue bar above your hotbar) should replenish instantly.
- Wait a little longer until a marketplace spawns, and press 'e' on the white gooey character walking around. Buy the empty cup from the shop.
- Return to the well and press 'e' while looking at the water. The cup should fill with water. Drink it by right clicking, and it should become empty again.
- Interact with the well by pressing 'e' until the tooltip reads "Drinks: 0/5". After this point, no thirst should be refilled by interacting with it again.
- Wait for a while when a well isn't at full capacity. After a bit of time, the well should replenish, one drink at a time.
